src/ai/core/signal_processor.py

```
@@ -0,0 +1,172 @@
+from dataclasses import dataclass
+import numpy as np
+
+@dataclass
+class SignalWeights:
+    intent: float = 0.35
+    market: float = 0.25
+    regional: float = 0.20
+    trends: float = 0.20
+
+class SignalProcessor:
+    def __init__(self, weights: Optional[SignalWeights] = None):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)